common/types.ts

```diff
@@ -19,6 +19,21 @@ export interface User {
   isAdmin: boolean;
   rsvpStatus: RSVPStatus;
 }
+export type SingletonDefinition = DateSingleton;
```

Comment:
Wasn't sure if this is what you meant lol

Reply:
yea, this is exactly what i meant. the only thing i'd change is to add the enum members explicitly, instead of just doing "date", so it would be something like this:

```typescript
export enum SingletonType {
  RegistrationOpen = 'registration-open',
  RegistrationClosed = 'registration-closed',
  ConfirmBy = 'confirm-by'
}

export interface DateSingleton {
  type: SingletonType.RegistrationOpen | SingletonType.RegistrationClosed | SingletonType.ConfirmBy;
}
```

and i think you can delete the ISingleton, i don't think you need it?